tests/ui/parser/missing-expression-in-for-loop.stderr

```diff
@@ -0,0 +1,13 @@
+error: expected expression, found block
+  --> $DIR/missing-expression-in-for-loop.rs:2:14
+   |
+LL |       for i in {
+   |  ______________^
+LL | |
+LL | |     }
+   | |_____^
+   |
+   = help: try adding an expression to the `for` loop: `for PATTERN in EXPRESSION`
```

Comment:
Could we turn this into a structured suggestion for the following? It would be easy to find the right span (`in_kw.span.between(block.span)`).

```
for i in /* expression */ {
```

Reply:
@obeis, you marked this as resolved but didn't implement this?

Reply:
What does he mean by `structured suggestion`?

Reply:
providing a suggestion, like:

```
help: try adding an expression to the `for` loop

| for i in /* expression */ {
           ++++++++++++++++
```

This is achieved with `#[suggestion(.....)]` in diagnostics derive.